src/main/java/org/folio/marc4ld/service/condition/ConditionCheckerImpl.java

```diff
@@ -141,12 +143,20 @@ private boolean isControlFieldConditions(List<ControlField> controlFields,
     return isEmpty(condition.getControlFields()) || condition.getControlFields().stream()
       .flatMap(cfc -> controlFields.stream()
         .filter(cf -> cf.getTag().equals(cfc.getTag()))
-        .map(cf -> {
-          var expression = expressionParser.parseExpression(cfc.getExpression());
-          cfc.setData(cf.getData());
-          return expression.getValue(new StandardEvaluationContext(cfc), Boolean.class);
-        }))
+        .map(cf -> evaluateControlField(cfc, cf)))
```

Comment:
MARC -> LD transformation fails when the length of the string in 008 is short (say, 10 characters). This fix (and also the changes in other classes) address this problem.